delocate/delocating.py

```diff
@@ -277,20 +330,33 @@ def delocate_path(tree_path, lib_path,
         is a file in the path depending on ``copied_lib_path``, and the value
         is the ``install_name`` of ``copied_lib_path`` in the depending
         library.
+
+    Raises
+    ------
+    DependencyNotFound
+        When any dependencies can not be located.
     """
     if lib_filt_func == "dylibs-only":
         lib_filt_func = _dylibs_only
+    elif isinstance(lib_filt_func, str):
+        raise TypeError('lib_filt_func string can only be "dylibs-only"')
+    lib_filt_func = lib_filt_func or (lambda _: True)
```

Comment:
Consider more explicit:

```suggestion
    lib_filt_func = lambda _ : True if lib_filt_func is None else lib_filt_func
```

Reply:
At that point the type was `Optional[Callable[[Text], bool]]` so I decided that simple truth testing was okay.

You put the lambda at the start and removed the parentheses around it so the ternary is now part of the lambda.  You also included an odd whitespace after the underscore.

If I was focusing solely on readability I'd go with this:
```python
if lib_filt_func is None:
    lib_filt_func = lambda _: True
```
But this version does a better job of replacing the `or` operator:
```python
lib_filt_func = lib_filt_func if lib_filt_func is not None else lambda _: True
```